### PR TITLE
Redraw messages obscured by liveview window

### LINKED ISSUE #15853: pixel minimap writes over live_view window
This occurs when the game window is tiny (e.g. default 80x24 size) and also occurs when the list of items on a square is very large. PR #15850 fixes the latter.

live_view in the tiny default window size (80x24). Notice the bottom half of the window is cut off by the pixel minimap.
before

For comparison, the lookview window is just fine.
fix

### LINKED ISSUE #16040: Message log hidden by expanded liveview window
When the liveview expands to accomodate longer descriptions, when it returns to a smaller size, part of the message log is black. The is caused by the way the various windows in the sidebar are refreshed.

I'm working on a PR to fix this.

Before using the liveview:
a

An expanded liveview:
b

After the liveview's size is reduced, a black region is left behind, obscuring the message log:
c

### PR DESCRIPTION
**Pull request by lucianposton**
This change fixes issue #16040 by rendering both the message log and liveview windows on each update, rather than only liveview.

~~Because the 2 windows overlap, the performance is very poor. Calling werase()+wrefresh() repeatedly between overlapping windows is terribly slow (even if both windows are entirely empty, which is still puzzling me).~~ There are a couple additional commits in the PR to make minor improvements to the performance~~, but it's still a significant performance regression in its current state.~~

~~The other idea I have is to put the liveview box inside the w_messages window. That'll completely get around the overlapping windows issues. I'll update this PR with to do just that in the next few days.~~ _Edit_: done

Commit 0250bfc removes the separate window for liveview, combining it into w_messages. Both now write into the same window, mitigating the performance issues of writing into overlapping windows. This also fixes #15853 by virtue of w_messages not overlapping the pixel minimap.

Fixes #16040
Fixes #15853

### CODE REVIEW

File: src/live_view.cpp [hunk 1]
```diff
@@ -40,76 +28,47 @@ void live_view::init( int const start_x, int const start_y, int const w, int con
...
+
+    // -1 for border. -1 because getmaxy() actually returns height, not y position.
+    const int last_line = height - 2;
+    const visibility_variables &cache = g->m.get_visibility_variables_cache();
+    bottom_border_line = START_LINE;
+    w_live_view->height = height;
+    g->print_all_tile_info( mouse_position, *this, START_COLUMN, bottom_border_line,
+                            last_line, false, cache );
 
+    w_live_view->height = std::max( bottom_border_line + 1, 11 );
```
Comment (BevapDin) [inline src/live_view.cpp]: You can't change a window size that way. `ncurses` windows don't have this property.

Comment (lucianposton) [inline src/live_view.cpp]: yea, I removed those #ifdef TILES checks. I added them back. I wanted to clear out all the unused code in live_view.cpp.

**Comment by lucianposton** [comment]
Removed [WIP] tag. Ready to merge.

**Comment by lucianposton** [comment]
out

**Pull request merged**